source/app/service-providers/cli-provider.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        cli-provider class
+ * CVM-Role:        Service Provider
+ * Authorr:         Felix Nüsse
+ * License:         GNU GPL v3
+ *
+ * Description:     This class handles the cli-arguments.
+ *                  It can be used to query arguments, as
+ *                  long as they are defined in the options object.
+ *
+ * END HEADER
+ */
+
+import { app } from 'electron'
+import path from 'path'
+
+export const DATA_DIR: string = 'data-dir'
+export const DISABLE_HARDWARE_ACCELERATION: string = 'disable-hardware-acceleration'
+export const CLEAR_CACHE: string = 'clear-cache'
+export const LAUNCH_MINIMIZED: string = 'launch-minimized'
+
+export function getCLIArgument (key: string): any {
+  switch (key) {
+    case DATA_DIR: {
+      let dataDir = getArgumentValue('--data-dir')
+      if (dataDir !== undefined && !path.isAbsolute(dataDir)) {
+        if (app.isPackaged) {
+          // Attempt to use the executable file's path as the basis
+          dataDir = path.join(path.dirname(app.getPath('exe')), dataDir)
+        } else {
+          // Attempt to use the repository's root directory as the basis
+          dataDir = path.join(__dirname, '../../', dataDir)
+        }
+        return dataDir
+      }
+      return undefined
+    }
+    case CLEAR_CACHE: {
+      return process.argv.includes('--clear-cache')
+    }
+    case DISABLE_HARDWARE_ACCELERATION: {
+      return process.argv.includes('--disable-hardware-acceleration')
+    }
+    case LAUNCH_MINIMIZED: {
+      return process.argv.includes('--launch-minimized') || process.argv.includes('-m')
+    }
+  }
+  return undefined
+}
+
+export function handleGeneralArguments (): void {
```

Comment:
This is a function that should only once be called by the `main.ts` entry point, correct? If so we should definitely state that, in comments. I just realized that we need Docstrings for the functions here

Reply:
I hope that is what you meant, i have added a comment explaining what it ~~does~~ should do

Reply:
Yes, now I see what is going on here. Please feel free to add DocBlocks to the other functions as well!

Reply:
I did that!